autoload/lsp/diag.vim

```diff
@@ -146,7 +146,7 @@ def DiagsRefreshSigns(lspserver: dict<any>, bnr: number)
                            text_align: 'above',
                            text_padding_left: diag.range.start.character})
       endif
-    catch /E966\|E964/ # Invalid lnum | Invalid col
+    catch /E966\|E964\|E475/ # Invalid lnum | Invalid col | Invalid arg
```

Comment:
Can you tell me in which case is "E475" raised?

Reply:
when option value is invalid, e.g end_col is 0

Reply:
The reason I ask is because I think the only scenario that E475 can be raised is iff the language server itself yields invalid ranges, which I think should be fixed in the language servers code, or at least be very explicitly handled in the client, with a reference to an issue on the specific language servers issue tracker.

Reply:
i think no, since actually not sure when or which one would be like that, but if it was invalid then prop hl would be disappered (removed) unexpectedly untill next upd, so had to try-catch it, unless we knew all invalid cases.

Reply:
> so had to try-catch it, unless we knew all invalid cases.

Can you provide code that can cause the "E475" error to happen? Because if not, then I think we shouldn't add more complexity than needed.

Reply:
um, seems i had offered an example? e.g end_col is 0

Reply:
> um, seems i had offered an example? e.g end_col is 0

When will that happen? Can you provide code that would result in a language server sending `diag.range.end == -1`? Remember that we add `+1`

Reply:
you knew that's just an example, right? there are not only one option end_col, and one condition, one lsp server, right?

Reply:
> The reason I ask is because I think the only scenario that E475 can be raised is iff the language server itself yields invalid ranges, which I think should be fixed in the language servers code, or at least be very explicitly handled in the client, with a reference to an issue on the specific language servers issue tracker.

Reply:
that's same for E966 and E964, why you added and catch that?
i think we sure would remove these ugly code later, but kept it working for now is required also, right?

Reply:
> that's same for E966 and E964, why you added and catch that?

I believe that there is a comment about those here https://github.com/yegappan/lsp/blob/8af226327289ea84c07415c2665875bde6a44931/autoload/lsp/diag.vim#L147